code_generator/End2End/Tests/NonBlittableComponentTests.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Generated.Improbable.TestSchema;
+using Generated.Improbable.TestSchema.Nonblittable;
+using NUnit.Framework;
+using Unity.Mathematics;
+using UnityEngine;
+
+namespace Improbable.Gdk.CodeGenerator.End2EndTests
+{
+    [TestFixture]
+    public class NonBlittableComponentTests
+    {
+        private static readonly bool1 Bool1False = new bool1(false);
+        private static readonly bool1 Bool1True = new bool1(true);
+
+        private const int IntValue = 123;
+        private const long LongValue = 5678L;
+        private const float FloatValue = 1.2345f;
+        private const double DoubleValue = 3.14159;
+        private static readonly bool1 BoolValue = Bool1True;
+        private const string StringValue = "A String value";
+
+        [Test]
+        public void component_should_subclass_unityengine_component()
+        {
+            var component = new SpatialOSNonBlittableComponent();
+            Assert.True(component is Component,
```

Comment:
Also check that its a `ISpatialComponentData`

Reply:
Done